File: GithubBrowserSample2/app/src/java/java/com/android/example/github/binding/FragmentBindingAdapters.java
package com.android.example.github.binding;

import android.widget.ImageView;


import javax.inject.Inject;

import androidx.fragment.app.Fragment;

/**
 * @author chenweiming
 * @version 1.0
 * @mail cwm930215@gmail.com
 * @since 2019/1/4 7:39 PM
 **/
public class FragmentBindingAdapters {

    private final Fragment fragment;

    @Inject
    public FragmentBindingAdapters(Fragment fragment){
        this.fragment = fragment;
    }

    public Fragment getFragment() {
        return this.fragment;
    }


    public void bindImage(ImageView imageView,String url){

    }



}

File: GithubBrowserSample3/app/src/main/java/com/android/example/github/binding/FragmentBindingAdapters.kt
package com.android.example.github.binding

import android.app.DownloadManager
import android.graphics.drawable.Drawable
import android.widget.ImageView
import androidx.databinding.BindingAdapter
import androidx.fragment.app.Fragment
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestListener
import javax.inject.Inject

/**
 * @author chenweiming
 * @version 1.0
 * @mail cwm930215@gmail.com
 * @since 2019/3/5 3:45 PM
 **/
class FragmentBindingAdapters @Inject constructor(val fragment: Fragment) {
    @BindingAdapter(value = ["imageUrl", "imageRequestListener"], requireAll = false)
    fun bindImage(imageView: ImageView, url: String?, listener: RequestListener<Drawable?>) {
        Glide.with(fragment).load(url).listener(listener).into(imageView)
    }


}